test_conformance/api/test_api_consistency.cpp

```diff
@@ -0,0 +1,478 @@
+            "not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMMemcpy(queue, CL_TRUE, ptr1, ptr0, allocSize, 0,
+                                   NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMMemcpy did not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ, ptr1, allocSize, 0,
+                                NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMMap did not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMUnmap(queue, ptr1, 0, NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMUnmap did not return CL_INVALID_OPERATION");
+
+        error = clFinish(queue);
+        test_error(error, "Error calling clFinish after SVM operations");
+
+        clSVMFree(context, ptr0);
+        error = clEnqueueSVMFree(queue, 1, &ptr1, NULL, NULL, 0, NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMFree did not return CL_INVALID_OPERATION");
+
+        error = clFinish(queue);
+        test_error(error, "Error calling clFinish after SVM free");
+
+        // clSetKernelArgSVMPointer, clSetKernelExecInfo
+        // Returns CL_INVALID_OPERATION if no devices in the context associated
+        // with kernel support Shared Virtual Memory.
+
+        error = clSetKernelArgSVMPointer(kernel, 0, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clSetKernelArgSVMPointer did not return CL_INVALID_OPERATION");
+
+        error =
+            clSetKernelExecInfo(kernel, CL_KERNEL_EXEC_INFO_SVM_PTRS, 0, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clSetKernelExecInfo did not return CL_INVALID_OPERATION");
+    }
+
+    return TEST_PASS;
+}
+
+static int check_atomic_capabilities(cl_device_atomic_capabilities atomicCaps,
+                                     cl_device_atomic_capabilities requiredCaps)
+{
+    if ((atomicCaps & requiredCaps) != requiredCaps)
+    {
+        log_error("Atomic capabilities %llx is missing support for at least "
+                  "one required capability %llx!\n",
+                  atomicCaps, requiredCaps);
+        return TEST_FAIL;
+    }
+
+    if ((atomicCaps & CL_DEVICE_ATOMIC_SCOPE_ALL_DEVICES) != 0
+        && (atomicCaps & CL_DEVICE_ATOMIC_SCOPE_DEVICE) == 0)
+    {
+        log_info("Check: ATOMIC_SCOPE_ALL_DEVICES is supported, but "
```

Comment:
The intention was only print info, no error ?

Reply:
Yes, this was intentional - it's not clear to me that this is strictly forbidden by the spec, though it is a little strange so I thought it was worth adding to the log.

Reply:
Hmm, we should probably create a spec issue for this.

Reply:
I've created https://github.com/KhronosGroup/OpenCL-Docs/issues/404 to track the spec question.  Suggest that we leave this as-is for now and I will update this check pending resolution of the spec issue.

Reply:
Works for me.